### ISSUE TITLE
unable to determine outcome based on API shape: NewSeed should return ([]byte, error).

### ISSUE THREAD

**Issue by sneak** [tyler-smith/go-bip39#37]
```
  seed := bip39.NewSeed(mnemonic, "Secret Passphrase")
```

This should return ([]byte, error) and validate the mnemonic.  Without reading your library, it's unclear what happens if you pass in a mnemonic that's the wrong number of words, or contains words not on the wordlist, et c.

**Comment by tyler-smith** [comment]
You are right, this is a bad API design. You should explicitly check that the mnemonic is valid with the provided utility functions. In v2 there will be no way to instantiate an invalid mnemonic without an error being presented.
